File: php/Problem9.php
<?php
/*
 * A Pythagorean triplet is a set of three natural numbers, a  b  c, for which,
 * a2 + b2 = c2
 * For example, 32 + 42 = 9 + 16 = 25 = 52.
 * There exists exactly one Pythagorean triplet for which a + b + c = 1000.
 * Find the product abc.
 */

for ($m = 0; $m < 23; $m++) {
    for ($n = 0; $n < $m; $n++) {
        $a = $m * $m - $n * $n;
        $b = 2 * $m * $n;
        $c = $m * $m + $n * $n;
        if ($a + $b + $c == 1000) {
            echo $a * $b * $c;
            break;
        }
    }
}

File: go/Problem9.go
/*
 * A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
 *
 * a^2 + b^2 = c^2
 *
 * For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.
 *
 * There exists exactly one Pythagorean triplet for which a + b + c = 1000.
 * Find the product abc.
 */
package main

import (
	"fmt"
	"math"
)

func main() {
	m := 1
	n := 1
	for ; m*(m+n) != 500; n++ {
		if m < n {
			m++
			n = 1
		}
	}
	answer := int(math.Pow(float64(m), 4)-math.Pow(float64(n), 4)) * 2 * m * n
	fmt.Println(answer)
}
